static/src/javascripts/es6/projects/common/modules/crosswords/main.js

```diff
@@ -5,32 +5,46 @@ import React from 'react';
 
 import $ from 'common/utils/$';
 import _ from 'common/utils/_';
+import detect from 'common/utils/detect';
+import scroller from 'common/utils/scroller';
 
 import Clues from './clues';
 import Controls from './controls';
-import FocussedClue from './focussed-clue.jsx!';
+import HiddenInput from './hidden-input';
 import Grid from './grid';
 import helpers from './helpers';
 import keycodes from './keycodes';
 import persistence from './persistence';
 import loadFont from './font';
 
-// make react available to dev tool
-if (!window.React) {
-    window.React = React;
-}
+class Crossword extends React.Component {
+
+    constructor (props) {
+        super(props);
 
-const Crossword = React.createClass({
-    getInitialState () {
         const dimensions = this.props.data.dimensions;
 
         this.columns = dimensions.cols;
         this.rows = dimensions.rows;
         this.clueMap = helpers.buildClueMap(this.props.data.entries);
 
+        this.onCheat = this.onCheat.bind(this);
```

Comment:
Oooh that is FUNKY

On 12 May 2015 at 14:46, Alex Sanders <notifications@github.com> wrote:

> In static/src/javascripts/es6/projects/common/modules/crosswords/main.js
> <https://github.com/guardian/frontend/pull/9144#discussion_r30134629>:
>
> >          const dimensions = this.props.data.dimensions;
> >
> >          this.columns = dimensions.cols;
> >          this.rows = dimensions.rows;
> >          this.clueMap = helpers.buildClueMap(this.props.data.entries);
> >
> > +        this.onCheat = this.onCheat.bind(this);
>
> agreed - was going to have a chat more generally about we do this, since
> will be common. esp wrt https://github.com/andreypopp/autobind-decorator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/9144/files#r30134629>.
>



-- 

"But I don't want to go among mad people," Alice remarked.

"Oh, you can't help that," said the Cat: "we're all mad here. I'm mad.
You're mad."

"How do you know I'm mad?" said Alice.

"You must be," said the Cat, "or you wouldn't have come here."

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396